### ISSUE TITLE
Fix FP S1172 ('no-unused-function-argument'): Ignore parameters referenced as JSX identifiers

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#3117]
S1172 currently raises false positives when a parameter occurs only in the position of a JSX identifier. This happens because ESLint context does not bind JSX identifier references to parameter definitions, if any.

```javascript
function f(Tag) { // FP: `Tag` is flagged as unused
  return <Tag />
}
```

We need to resolve those references ourselves and filter out "unused" parameters if they are actually used as JSXIdentifier nodes.

https://community.sonarsource.com/t/cannot-suppress-jsx-false-positive-javascript-s1172-unusedfunctionargument-in-intellij-idea-ultimate/61474
